src/main/java/org/folio/rest/utils/MigrationUtils.java

```diff
@@ -0,0 +1,164 @@
+package org.folio.rest.utils;
+
+import static org.folio.rest.impl.InvoiceStorageImpl.INVOICE_TABLE;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.common.ModuleId;
+import org.folio.okapi.common.SemVer;
+import org.folio.rest.core.RestClient;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.core.models.RequestEntry;
+import org.folio.rest.jaxrs.model.Invoice;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.CriterionBuilder;
+import org.folio.rest.persist.DBClient;
+
+public class MigrationUtils {
+  private static final Logger log = LogManager.getLogger(HelperUtils.class);
+
+  private MigrationUtils() {
+  }
+
+  public static Future<Void> migration(TenantAttributes attributes, String migrationModule, Supplier<Future<Void>> supplier) {
+    if (StringUtils.isBlank(attributes.getModuleFrom()))
+      return Future.succeededFuture();
+
+    SemVer moduleTo = moduleVersionToSemVer(migrationModule);
+    SemVer currentModuleVersion = moduleVersionToSemVer(attributes.getModuleFrom());
+    if (moduleTo.compareTo(currentModuleVersion) > 0) {
+      return supplier.get();
+    }
+    return Future.succeededFuture();
+  }
+
+  public static SemVer moduleVersionToSemVer(String version) {
+    try {
+      return new SemVer(version);
+    } catch (IllegalArgumentException ex) {
+      return new ModuleId(version).getSemVer();
+    }
+  }
+
+  public static Future<Void> populateInvoiceWithFiscalYear(Map<String, String> headers, Context vertxContext) {
+    Promise<Void> promise = Promise.promise();
+    vertxContext.runOnContext(event -> {
+      DBClient client = new DBClient(vertxContext, headers);
+      RequestContext requestContext = new RequestContext(vertxContext, headers);
+      client.startTx()
```

Comment:
Thanks to notice, used Conn

Reply:
one minor improvement, lets close ')' for .withTrans() method before .onSuccess(), .onFailure()
Basically it will work in current implementation, but better I think to write about results of operation after transaction completes

Reply:
Fixed